tests/models/templates/test_csms_api.py

```diff
@@ -21,13 +24,335 @@
+                new_manifest["samples"] = [
+                    {
+                        k: v if k != "protocol_identifier" else "bar"
+                        for k, v in sample.items()
+                    }
+                    for sample in manifest["samples"]
+                ]
+                detect_manifest_changes(new_manifest, user=user)
+            # this is why we needed a second valid trial to test this check
+            with pytest.raises(Exception, match="Change in critical field for"):
+                # CIDC trial_id = CSMS protocol_identifier
+                # stored on samples, not manifest
+                new_manifest = {k: v for k, v in manifest.items() if k != "samples"}
+                new_manifest["samples"] = [
+                    {
+                        k: v if k != "protocol_identifier" else "foo"
+                        for k, v in sample.items()
+                    }
+                    for sample in manifest["samples"]
+                ]
+                detect_manifest_changes(new_manifest, user=user)
+
+            # manifest_id has no such complication, but is also on the samples
+            # changing the manifest_id makes it new
+            with pytest.raises(NewManifestError):
+                new_manifest = {
+                    k: v if k != "manifest_id" else "foo" for k, v in manifest.items()
+                }
+                new_manifest["samples"] = [
+                    {k: v if k != "manifest_id" else "foo" for k, v in sample.items()}
+                    for sample in new_manifest["samples"]
+                ]
+                detect_manifest_changes(new_manifest, user=user)
+
+            # Changing a cimac_id is adding/removing a Sample
+            ## so this is a different error
+            with pytest.raises(Exception, match="Malformatted cimac_id"):
+                new_manifest = {k: v for k, v in manifest.items()}
+                new_manifest["samples"] = [
+                    {k: v if k != "cimac_id" else "foo" for k, v in sample.items()}
+                    if n == 0
+                    else sample
+                    for n, sample in enumerate(manifest["samples"])
+                ]
+                detect_manifest_changes(new_manifest, user=user)
+            # need to use an actually valid cimac_id
+            with pytest.raises(Exception, match="Missing sample"):
+                new_manifest = {k: v for k, v in manifest.items() if k != "samples"}
+                new_manifest["samples"] = [
+                    {
+                        k: v if k != "cimac_id" else "CXXXP0555.00"
+                        for k, v in sample.items()
+                    }
+                    if n == 0
+                    else sample
+                    for n, sample in enumerate(manifest["samples"])
+                ]
+                detect_manifest_changes(new_manifest, user=user)
+
+            # Test non-critical changes on the manifest itself
```

Comment:
I would stash these for a future sprint/PR lets just completely ignore the non-critical fields at this point.

Reply:
see https://github.com/CIMAC-CIDC/cidc-api-gae/pull/587#discussion_r722595714